packages/beachball/src/fixtures/repository.ts

```diff
@@ -0,0 +1,118 @@
+import { exec as nativeExec } from "child_process";
+import * as process from "process";
+import * as tmp from "tmp";
+
+async function dirAsync(options: tmp.DirOptions): Promise<tmp.DirResult> {
+  return new Promise((resolve, reject) => {
+    tmp.dir(options, (err, name, removeCallback) => {
+      if (err) {
+        reject(err);
+      } else {
+        resolve({ name, removeCallback });
+      }
+    });
+  });
+}
+
+interface PsResult {
+  stderr: string;
+  status: Error | null;
+  stdout: string;
+}
+
+function exec(command: string): Promise<PsResult> {
+  return new Promise(function(resolve, reject) {
+    nativeExec(command, (status, stdout, stderr) => {
+      const result = {
+        stderr,
+        stdout,
+        status
+      };
+      if (status) {
+        reject(result);
+      } else {
+        resolve(result);
+      }
+    });
+  });
+}
+
+async function runCommands(commands: string[]): Promise<void> {
+  for (let i = 0; i < commands.length; i++) {
+    await exec(commands[i]);
+  }
+}
+
+async function runInDirectory(targetDirectory: string, commands: string[]) {
+  const originalDirectory = process.cwd();
+  process.chdir(targetDirectory);
+  await runCommands(commands);
+  process.chdir(originalDirectory);
+}
+
+export class RepositoryFactory {
+  root?: tmp.DirResult;
+
+  async create(): Promise<void> {
+    this.root = await dirAsync({ prefix: "beachball-repository-upstream-" });
+    await runInDirectory(this.root!.name, [
+      "git init",
+      "touch README",
+      "git add README",
+      'echo \'{ "name": "foo" }\' > package.json',
```

Comment:
does this work across all platforms?

Reply:
I am guessing this will not work on windows.